pkg/iac/scanners/terraform/parser/parser_test.go

```diff
@@ -2187,3 +2187,35 @@ resource "aws_s3_bucket" "example" {
 	val := modules.GetResourcesByType("aws_s3_bucket")[0].GetAttribute("bucket").GetRawValue()
 	assert.Nil(t, val)
 }
+
+func Test_AttrIsRefToOtherBlock(t *testing.T) {
+	fsys := testutil.CreateFS(t, map[string]string{
+		"main.tf": `locals {
+  baz_idx = 0
+}
+
+resource "foo" "bar" {
+  attr = baz.qux[local.baz_idx].attr
+}
+
+resource "baz" "qux" {
+  count = 1
+  attr  = "test"
+}
+`,
+	})
+
+	parser := New(fsys, "", OptionStopOnHCLError(true))
+	require.NoError(t, parser.ParseFS(context.TODO(), "."))
+
+	modules, _, err := parser.EvaluateAll(context.TODO())
+	require.NoError(t, err)
+
+	require.Len(t, modules, 1)
+	root := modules[0]
+	foo := root.GetResourcesByType("foo")[0]
+	fooAttr := foo.GetAttribute("attr")
+	b, err := root.GetReferencedBlock(fooAttr, foo)
+	require.NoError(t, err)
+	require.NotNil(t, b)
```

Comment:
Maybe we can strengthen this assertion?

```suggestion
	assert.Equal(t, "test", b.Values().AsValueSlice()[0].AsString())
```